clang/lib/CConv/ProgramInfo.cpp

```diff
@@ -751,17 +754,59 @@ ProgramInfo::getStaticFuncConstraintSet(std::string FuncName,
   return nullptr;
 }
 
-bool ProgramInfo::computePointerDisjointSet() {
-  ConstraintDisjointSet.Clear();
-  CVars WildPtrs;
+class ReachableVarVisitor : public boost::default_bfs_visitor {
+public:
+  ReachableVarVisitor(const ConstraintKey &SrcWAtom,
+                      CVars &T,
+                      std::map<ConstraintKey, CVars> &RCMap,
+                      std::set<Atom *> &AllDWild):  SrcA(SrcWAtom),
+                      RootCauseMap(RCMap), VisitedCVars(T),
+                      AllDirectWild(AllDWild) { }
+  template < typename Vertex, typename Graph >
+  void discover_vertex(Vertex u, const Graph &g) const {
+    Atom *V = g[u];
+    VarAtom *VA = dyn_cast<VarAtom>(V);
+    if (VA != nullptr && AllDirectWild.find(VA) == AllDirectWild.end()) {
+      RootCauseMap[VA->getLoc()].insert(SrcA);
+      VisitedCVars.insert(VA->getLoc());
+    }
+  }
+  const ConstraintKey &SrcA;
+  std::map<ConstraintKey, CVars> &RootCauseMap;
+  CVars &VisitedCVars;
+  std::set<Atom *> &AllDirectWild;
+};
+
+bool ProgramInfo::computeInterimConstraintState() {
```

Comment:
Can you explain what this is, and what the invariants are about it at any given time?

Reply:
To repeat: This seems like the key functionality that's changed, but I don't know what it does. More help please!

Reply:
Added.